src/pp-annotator/form/AnnotationForm.tsx

```diff
@@ -125,15 +127,23 @@ export default class AnnotationForm extends React.Component<
     this.setState({ isLinkOnly: e.currentTarget.checked });
   }
 
-  private onSave() {
+  private onSave(event: any) {
     const fieldsToSave = getFormState(this.state);
     if (this.state.isLinkOnly) {
       fieldsToSave.comment = '';
     }
-    this.props.onSave(fieldsToSave);
+    this.props.annotation.fields = fieldsToSave;
+    const result = this.props.saveAction(this.props.annotation);
+    const errors = result.errors;
```

Comment:
This will not work when `saveAction` is an actual method that saves the annotation in the backend!

Anything interacting with the backend will not block and return results synchronously — rather, it will synchronously return a Promise with the results of the asynchronous action (communication with the backend). `saveAction` should really have a `Promise<T>` return type. This is also how the `create()` actions is defined in the Annotator [API docs](http://docs.annotatorjs.org/en/latest/api/storage.html#annotator.storage.StorageAdapter.prototype.create).

Reply:
Is what you suggest something like
```js
this.props.saveAction(this.props.annotation).then(
\\ success; close form 
).catch(
\\ render validation errors if they are the reason
)
```

or simply
```js
this.props.saveAction(this.props.annotation).then(
(result) => {
  if (result.errors)  {
\\ render validation errors
}
else {
\\ success; close form 
}

)
```

Reply:
This would depend on how saveAction is written, what the possible errors can be and whether they are treated as exceptional behaviour (shouldn't happen) and promise rejected, or a negative result (can happen in a normal flow) and promise resolved. So if you're going to use the method from Annotator it depends on what is does.